Assets/Scripts/Auction/AuctionDriver.cs

```diff
@@ -41,38 +47,73 @@ private void Start()
 #if UNITY_EDITOR
         biddingBeginDelay = 2f;
 #endif
-        availableAuctionStages = new RandomisedAuctionStage[] { StaticInfo.Singleton.BodyAuction, StaticInfo.Singleton.BarrelAuction, StaticInfo.Singleton.ExtensionAuction };
+        if (MatchController.Singleton.RoundCount == 1)
+            availableAuctionStages = new WeightedRandomisedAuctionStage[] { StaticInfo.Singleton.BodyAuction };
+        if (MatchController.Singleton.RoundCount == 2)
+            availableAuctionStages = new WeightedRandomisedAuctionStage[] { StaticInfo.Singleton.BarrelAuction };
+        if (MatchController.Singleton.RoundCount == 3)
+            availableAuctionStages = new WeightedRandomisedAuctionStage[] { StaticInfo.Singleton.ExtensionAuction };
+        if (MatchController.Singleton.RoundCount > 3)
+            availableAuctionStages = new WeightedRandomisedAuctionStage[] { StaticInfo.Singleton.BodyAuction, StaticInfo.Singleton.BarrelAuction, StaticInfo.Singleton.ExtensionAuction };
```

Comment:
Should this be a switch-case instead?

Reply:
Yep, the C# case shorthand makes it less lines of codes and more readable, will update